From ca826d155ba9b6126d7abd41895e5a9f4067241f Mon Sep 17 00:00:00 2001
From: Guido Falsi <mad@madpilot.net>
Date: Wed, 3 Apr 2024 16:07:11 +0200
Subject: [PATCH] Use correct literal suffix for `size_type`.

`std::vector<T,Allocator>::size` returns `size_type`. `%ld` works
fine on 64 bit arches, but fails on 32 bit.

Ref.: https://en.cppreference.com/w/cpp/container/vector/size
      https://en.cppreference.com/w/cpp/types/size_t
---
 gnucash/import-export/import-main-matcher.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gnucash/import-export/import-main-matcher.cpp b/gnucash/import-export/import-main-matcher.cpp
index 8a234d3bc2..8f7c18b751 100644
--- a/gnucash/import-export/import-main-matcher.cpp
+++ b/gnucash/import-export/import-main-matcher.cpp
@@ -847,7 +847,7 @@ gnc_gen_trans_assign_transfer_account_to_selection_cb (GtkMenuItem *menuitem,
     bool is_selection = true;
     auto debugging_enabled{qof_log_check (G_LOG_DOMAIN, QOF_LOG_DEBUG)};
 
-    DEBUG("Rows in selection = %ld", selected_refs.size());
+    DEBUG("Rows in selection = %zu", selected_refs.size());
 
     for (const auto& ref : selected_refs)
     {
-- 
2.44.0

